neurodamus/core/_utils.py

```diff
@@ -67,8 +69,44 @@ def rank0_wrapper(*args, **kw):
         return rank0_wrapper
 
 
+class SimulationProgress:
```

Comment:
Does this need to be a decorator? Wouldn't a plain class simplify it? It can be more easily tested, can be instantiated in `_run_neuron`, etc?

A bare function could work, too, by passing `last_time_check`, which would simplify it even more, I think.

Reply:
initially i had a simple class instantiated in _run_neuron as you mentioned, but i thought this way was more elegant, and the only difference is the __call__

I could leave the plain class if you prefer